### PR TITLE
bump 0ad

### PR DESCRIPTION
**Pull request by holgersson32644**
- dev-util/premake: Bump to 5.0.0_alpha16
- games-strategy/0ad: Bump to 0.0.25_alpha, unbundle premake
- games-strategy/0ad: Remove the old pre-alpha-24(!) version

### CODE REVIEW

File: dev-util/premake/premake-5.0.0_alpha16.ebuild [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MY_P=${P/_/-}
```
Comment (ionenwks) [inline dev-util/premake/premake-5.0.0_alpha16.ebuild:6]: This doesn't seem used.

File: dev-util/premake/premake-5.0.0_alpha16.ebuild [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+DESCRIPTION="A makefile generation tool"
+HOMEPAGE="https://premake.github.io/"
+SRC_URI="https://github.com/${PN}/${PN}-core/archive/v${MY_PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="BSD"
+
+SLOT=$(ver_cut 1)
+
+KEYWORDS="~amd64 ~ppc ~x86"
+
+S="${WORKDIR}/${PN}-core-${MY_PV}"
```
Comment (ionenwks) [inline dev-util/premake/premake-5.0.0_alpha16.ebuild:18]: Could use some re-ordering to match skel.ebuild and less spacing, and I'd argue this ver_cut isn't worth.. full SLOT aren't auto-changed on a whim and better to make it clear.
```suggestion
S="${WORKDIR}/${PN}-core-${MY_PV}"

LICENSE="BSD"
SLOT="5"
KEYWORDS="~amd64 ~ppc ~x86"
```

File: dev-util/premake/premake-5.0.0_alpha16.ebuild [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MY_P=${P/_/-}
+MY_PV=${PV/_/-}
+
+DESCRIPTION="A makefile generation tool"
+HOMEPAGE="https://premake.github.io/"
+SRC_URI="https://github.com/${PN}/${PN}-core/archive/v${MY_PV}.tar.gz -> ${P}.tar.gz"
```
Comment (ionenwks) [inline dev-util/premake/premake-5.0.0_alpha16.ebuild:11]: First two `${PN}` could be plaintexted.
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Plaintexting_.24.7BPN.7D_in_SRC_URI

**Review by ionenwks — COMMENTED** [review]
Just getting the usual nitpicks out of the way for premake.

**Comment by thesamesam** [comment]
Thanks!

**Pull request closed without merging**